src/go/k8s/pkg/resources/configmap.go

```diff
@@ -422,24 +425,45 @@ func (r *ConfigMapResource) preparePandaproxyClient(
 	return nil
 }
 
-func (r *ConfigMapResource) createBasicAuthSecret(
-	ctx context.Context, username, password string,
+func (r *ConfigMapResource) prepareSchemaRegistryClient(
+	ctx context.Context, cfgRpk *config.Config,
 ) error {
-	key := r.keySASL()
-	obj := &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      key.Name,
-			Namespace: key.Namespace,
-		},
-		Type: corev1.SecretTypeBasicAuth,
-		Data: map[string][]byte{
-			corev1.BasicAuthUsernameKey: []byte(username),
-			corev1.BasicAuthPasswordKey: []byte(password),
-		},
+	if r.pandaCluster.Spec.Configuration.SchemaRegistry == nil {
+		return nil
+	}
+
+	replicas := *r.pandaCluster.Spec.Replicas
+	cfgRpk.SchemaRegistryClient = &config.KafkaClient{}
+	for i := int32(0); i < replicas; i++ {
+		cfgRpk.SchemaRegistryClient.Brokers = append(cfgRpk.SchemaRegistryClient.Brokers, config.SocketAddress{
+			Address: fmt.Sprintf("%s-%d.%s", r.pandaCluster.Name, i, r.serviceFQDN),
+			Port:    r.pandaCluster.InternalListener().Port,
```

Comment:
the external settings does not have to be reflected here?

Reply:
The schema registry client is ment for externalize schema registry from the Redpanda container. In theory if schema registry is running inside Redpanda process it should auto discover this configuration. I copy the same snippet of configuration from panda proxy to be consistent. 

If we would like to have multiple containers within Redpanda POD we need to be explicit about how schema registry should talk to Redpanda. The auto discovery feature will not work if schema registry is running out of process.

Reply:
Auto discovery also doesn't work for SASL, only broker TLS.